src/monitor-location-workspace/monitor-location-checks.service.ts

```diff
@@ -37,11 +37,9 @@ export class MonitorLocationChecksService {
     const locations: LocationIdentifiers[] = [];
 
     const addLocation = (i: ForLocationType) => {
-      const id = i?.unitId || i?.stackPipeId;
-      let location = locations.find(l => {
-        const locationId = l?.unitId || l?.stackPipeId;
-        return locationId === id;
-      });
+      let location = locations.find(
+        l => l?.unitId === i?.unitId || l?.stackPipeId === i?.stackPipeId,
+      );
```

Comment:
This change was meant to fix a bug with not all locations being found. I think this is a merge overwriting something it shouldn't.